vnext/Scripts/msbuildeval.ps1

```diff
@@ -0,0 +1,73 @@
+# Evaluate an MSBuild property name
+param(
+    [Parameter(Mandatory=$true)]
+    [String]$SolutionFile,
+    [Parameter(Mandatory=$true)]
+    [String]$ProjectFile,
+    [Parameter(Mandatory=$true)]
+    [String]$PropertyName
+)
+
+function Get-MSBuildPath {
+    $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
+    if (!(Test-Path $vsWhere)) {
+        throw "Unable to find vswhere.exe."
+    }
+    $vsPath = & $vsWhere -version 16.5 -property installationPath;
+    return "$vsPath\MSBuild\Current\Bin\";
+}
+
+function Get-MSBuildProperty {
+    param (
+        [Parameter(Mandatory=$true)]
+        [String]$SolutionPath,
+        [Parameter(Mandatory=$true)]
+        [String]$ProjectPath,
+        [Parameter(Mandatory=$true)]
+        [String]$PropertyName
+    )
+
+    # Identify location of MSBuild
+    $MsBuildPath = Get-MSBuildPath
+
+    # Load Microsoft.Build.dll into script
+    Add-Type -Path "$MsBuildPath\Microsoft.Build.dll" | Out-Null
+
+    # Build a temporary "metaproj" of the solution file so it can be processed
+    ${env:MSBUILDEMITSOLUTION} = 1
+    & $MsBuildPath\MSBuild.exe $SolutionPath | Out-Null
+
+    # Build a local project collection
+    $projectCollection =[Microsoft.Build.Evaluation.ProjectCollection]::new()
+
+    # Process solution
+    $solution = [Microsoft.Build.Evaluation.Project]::new("$SolutionPath.metaproj", $null, "Current", $projectCollection)
```

Comment:
What exactly does `Process solution` mean?
Does it run the default (usually `Build`) target?
If so, why deal with `.metaproj`?

Reply:
.sln files are a very archaic file format. There have been many attempts to remove it but it never happend. Since .sln files have very strict control over configuration, i.e it can override that project Abc will build it's x86 when the solutoin is build using arm64 configuration. It can configure to skip building certain projects in certain configurations etc etc..

For MsBuild to handle this it actually generates an msbuild file under the covers that implements that special logic. This is the metaproj file. If you run a build using `/bl` you can actually see the engine building that metaproj and it is driving the build of the projects inthe .sln file.